src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/Ncwms.java

```diff
@@ -182,41 +175,51 @@ private String encodeMapForRequest(Map<String, String[]> params) {
         return combinedStyles;
     }
 
-    public static List<String> getSupportedStyles(Document getCapabilitiesXml, String layerName) {
-        Set<String> styles = new HashSet<>();
-        for (final String combinedStyle : getCombinedStyles(getCapabilitiesXml, layerName)) {
-            styles.add(combinedStyle.split("/")[0]);
-        }
-        return new ArrayList<>(styles);
-    }
-
-    public static List<String> getPalettes(Document getCapabilitiesXml, String layerName) {
-        Set<String> palettes = new HashSet<>();
-        for (final String combinedStyle : getCombinedStyles(getCapabilitiesXml, layerName)) {
-            palettes.add(combinedStyle.split("/")[1]);
-        }
-        return new ArrayList<>(palettes);
-    }
-
     private Document getCapabilitiesXml(LayerDescriptor layerDescriptor)
         throws IOException {
         String wmsUrl = null;
         try {
             wmsUrl = getWmsUrl(layerDescriptor, null);
             String getCapabilitiesUrl =
                 String.format("%s?service=WMS&version=1.3.0&request=GetCapabilities",
-                    wmsUrl
+                        wmsUrl
                 );
 
             SAXReader reader = new SAXReader();
             return reader.read(new URL(getCapabilitiesUrl));
-        }
-        catch (DocumentException e) {
+        } catch (DocumentException e) {
             LOGGER.log(Level.SEVERE, String.format("Error parsing GetCapabilities XML document at '%s'", wmsUrl));
             throw new IOException(e);
         }
     }
 
+    public org.json.simple.JSONObject getMetadataJson(LayerDescriptor layerDescriptor) {
+        Object obj = null;
+        String wmsUrl = null;
+
+        try {
+            wmsUrl = getWmsUrl(layerDescriptor, null);
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, String.format("Error retreiving getWmsUrl: '%s'", e.getStackTrace()));
+            e.printStackTrace();
+        }
+
+        String metadataUrl = String.format("%s?service=WMS&version=1.3.0&request=GetMetadata&item=layerDetails&layerName=%s",
+                wmsUrl,
+                layerDescriptor.getVariableName()
+        );
+
+        JSONParser parser = new JSONParser();
```

Comment:
JSONParser implements Closable. 

Would it make sense to initialize it in a try clause, to prevent resource leaks?
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html